File: test/fixtures/output/ocaml/cohttp/multipart-data.ml
open Cohttp_lwt_unix
open Lwt

let uri = Uri.of_string "http://mockbin.com/har" in
let headers = Header.init ()
  |> fun h -> Header.add h "content-type" "multipart/form-data; boundary=---011000010111000001101001"
in
let body = "-----011000010111000001101001\r\nContent-Disposition: form-data; name=\"foo\"; filename=\"hello.txt\"\r\nContent-Type: text/plain\r\n\r\nHello World\r\n-----011000010111000001101001--" in

Client.call ~headers ~body (Code.method_of_string "POST") uri
>>= fun (res, body_stream) ->
  (* Do stuff with the result *)

File: test/fixtures/output/node/fetch/multipart-data.js
var fs = require("fs");
var FormData = require("form-data");
var fetch = require("node-fetch");

var formData = new FormData();

formData.append("foo", fs.createReadStream("hello.txt"));

var url = 'http://mockbin.com/har';

var options = {
  method: 'POST',
  headers: {'content-type': 'multipart/form-data; boundary=---011000010111000001101001'}
};

options.body = formData;

fetch(url, options)
  .then(res => res.json())
  .then(json => console.log(json))
  .catch(err => console.error('error:' + err));
